bundles/org.openhab.binding.siemenshvac/src/main/java/org/openhab/binding/siemenshvac/internal/Metadata/SiemensHvacMetadataDataPoint.java

```diff
@@ -0,0 +1,241 @@
+        this.dptId = dptId;
+    }
+
+    public int getDptSubKey() {
+        return dptSubKey;
+    }
+
+    public void setDptSubKey(int dptSubKey) {
+        this.dptSubKey = dptSubKey;
+    }
+
+    public @Nullable String getAddress() {
+        return address;
+    }
+
+    public void setWriteAccess(boolean writeAccess) {
+        this.writeAccess = writeAccess;
+    }
+
+    public boolean getWriteAccess() {
+        return writeAccess;
+    }
+
+    public void setAddress(String address) {
+        this.address = address;
+    }
+
+    public @Nullable String getDptUnit() {
+        return dptUnit;
+    }
+
+    public void setDptUnit(String dptUnit) {
+        this.dptUnit = dptUnit;
+    }
+
+    public @Nullable String getMaxLength() {
+        return maxLength;
+    }
+
+    public void setMaxLength(String maxLength) {
+        this.maxLength = maxLength;
+    }
+
+    public @Nullable String getDialogType() {
+        return dialogType;
+    }
+
+    public void setDialogType(String dialogType) {
+        this.dialogType = dialogType;
+    }
+
+    public @Nullable String getMin() {
+        return min;
+    }
+
+    public void setMin(String min) {
+        this.min = min;
+    }
+
+    public @Nullable String getMax() {
+        return max;
+    }
+
+    public void setMax(String max) {
+        this.max = max;
+    }
+
+    public @Nullable String getResolution() {
+        return resolution;
+    }
+
+    public void setResolution(String resolution) {
+        this.resolution = resolution;
+    }
+
+    public @Nullable String getFieldWitdh() {
+        return fieldWitdh;
+    }
+
+    public void setFieldWitdh(String fieldWitdh) {
+        this.fieldWitdh = fieldWitdh;
+    }
+
+    public @Nullable String getDecimalDigits() {
+        return decimalDigits;
+    }
+
+    public void setDecimalDigits(String decimalDigits) {
+        this.decimalDigits = decimalDigits;
+    }
+
+    public Boolean getDetailsResolved() {
+        return detailsResolved;
+    }
+
+    public void setDetailsResolved(Boolean detailsResolved) {
+        this.detailsResolved = detailsResolved;
+    }
+
+    public void resolveDptDetails(JsonObject result) {
+        JsonObject subResultObj = result.getAsJsonObject("Result");
+        JsonObject desc = result.getAsJsonObject("Description");
+
+        if (subResultObj.has("Success")) {
+            JsonObject error = subResultObj.getAsJsonObject("Error");
+            String errorMsg = "";
+            if (error != null) {
+                errorMsg = error.get("Txt").getAsString();
+            }
+
+            if (("datatype not supported").equals(errorMsg)) {
+                detailsResolved = true;
+                unresolvableDetails = true;
+                return;
+            }
+
+        }
+
+        if (desc != null) {
+            this.dptType = desc.get("Type").getAsString();
+
+            if (dptType.equals(SiemensHvacBindingConstants.DPT_TYPE_ENUM)) {
```

Comment:
Better : `SiemensHvacBindingConstants.DPT_TYPE_ENUM.equals(dptType)`

Reply:
ok fixed in next commit